homework/index.js

```diff
@@ -30,18 +32,116 @@
+      class: 'li',
+    });
+    createAndAppend('li', listRepoInfo, { text: `Forks: ${selectedRepo.forks}`, class: 'li' });
+    createAndAppend('li', listRepoInfo, {
+      text: `Updated: ${selectedRepo.updated_at}`,
+      class: 'li',
+    });
+  }
+
+  // create (right box) as a table to render contributor Info
+  function createContributor(selectedRepo, repoContainer, root) {
+    const contributorUrl = selectedRepo.contributors_url;
+    fetchJSON(contributorUrl).then(contributors => {
+      const table = createAndAppend('table', repoContainer, { class: 'table' });
+
+      const tBody = createAndAppend('tbody', table);
+
+      contributors
+        .forEach(contributor => {
+          const row = createAndAppend('tr', tBody);
+
+          const imageCell = createAndAppend('td', row);
+          createAndAppend('img', imageCell, {
+            src: contributor.avatar_url,
+            class: 'img',
+            alt: 'contributor personal photo',
+          });
+          const nameCell = createAndAppend('td', row);
+          createAndAppend('a', nameCell, {
+            target: '_blank',
+            href: contributor.html_url,
+            text: contributor.login,
+            class: 'contributor-name',
+          });
+          const contributionsNum = createAndAppend('td', row);
+          createAndAppend('span', contributionsNum, {
+            text: contributor.contributions,
+            class: 'contributor-number',
+          });
+        })
+        .catch(err => {
+          createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+        });
+    });
+  }
+
   function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
+    const root = document.getElementById('root');
+    fetchJSON(url)
+      .then(data => {
+        data.sort((a, b) => a.name.localeCompare(b.name));
+
+        const header = createAndAppend('header', root, { class: 'header' });
+
+        createAndAppend('p', header, { text: 'HYF Repositories' });
+
+        const select = createAndAppend('select', header, { id: 'select' });
+
+        data.forEach((repository, index) => {
+          createAndAppend('option', select, { text: repository.name, value: index });
+        });
+
+        // add event change so we can remove the displayed repo info when we change the name of the repo
+        select.addEventListener('change', event => {
+          // get repo container (parent) and empty its content
+          const repoContainer = document.getElementsByClassName('repo-container')[0];
+          removeChildren(repoContainer);
+
+          // getting the selected repo using the value of event.target which is select element
+          const selectedRepo = data[event.target.value];
+
+          // rendering info of repository which we chose
+          createRepoInfo(selectedRepo, repoContainer);
+          createContributor(selectedRepo, repoContainer, root);
+        });
```

Comment:
You could move all of this code (lines 120-131) just before line 139. If you do that you don't need line 122 to get a reference to the `repoContainer` as you have it already from line 133.

Reply:
Thanks a lot! that is handy. It's done! :)